cpp-client/build-dependencies.sh

```diff
@@ -561,24 +562,25 @@ if [ "$CLONE_PROTOBUF" = "yes" ]; then
   echo
   echo "*** Cloning protobuf"
   cd $SRC
-  # Previously used version: v3.21.12
-  git clone $GIT_FLAGS -b v3.21.2 --depth 1 "${GITHUB_BASE_URL}/protocolbuffers/protobuf.git"
+  # Previously used version: v3.21.2
+  git clone $GIT_FLAGS -b v25.3 --depth 1 "${GITHUB_BASE_URL}/protocolbuffers/protobuf.git"
```

Comment:
1. Were we ultra out of date?
2. Did they change their versioning convention?
3. Do we need to specify the patch version?